### ISSUE TITLE
Bert4Rec 预测

### ISSUE THREAD

**Issue by youngzw** [RUCAIBox/RecBole#1183]
https://github.com/RUCAIBox/RecBole/blob/7236c07b1b0e039814e168ef8926dcba85a9b26c/recbole/model/sequential_recommender/bert4rec.py#L228-L247

你好，请问Bert4Rec这里最后为什么计算score呀，原论文不是用的概率吗？

**Comment by youngzw** [comment]
原论文优化目标是物品概率

**Comment by leoleojie** [comment]
@youngzw 您好！感谢您对伯乐的关注，predict函数是为了计算候选商品的得分，从而进行后续的性能评估。
BERT4Rec使用序列末尾添加的[MASK]的隐层表示来预测下一个商品。
![image](https://user-images.githubusercontent.com/88081427/157714155-8ef4eba0-a026-4800-871f-f0a556017f90.png)
而原论文所使用的预测概率就是将隐层表示与item embedding相乘，来计算item的得分，与我们的实现是一致的。

**Comment by youngzw** [comment]
https://github.com/RUCAIBox/RecBole/blob/7236c07b1b0e039814e168ef8926dcba85a9b26c/recbole/model/sequential_recommender/bert4rec.py#L216-L226

谢谢你的回复，上面的截图有一步与E^T相乘，但是后面还有一步softmax操作，我在损失函数这里好像也没看到类似的操作。

**Comment by youngzw** [comment]
```
# training settings
epochs: 500 #训练的最大轮数
train_batch_size: 4096 #训练的batch_size
learner: adam #使用的pytorch内置优化器
learning_rate: 0.001 #学习率
training_neg_sample_num: 0 #负采样数目
eval_step: 1 #每次训练后做evalaution的次数
stopping_step: 10 #控制训练收敛的步骤数，在该步骤数内若选取的评测标准没有什么变化，就可以提前停止了
```
你好，我在看RecBole教程的时候看到有个`eval_step: 1`参数和`stopping_step: 10`参数，后面的`stopping_step: 10`是早停法的步数，但是这个`eval_step: 1`参数后面备注是`#每次训练后做evalaution的次数`，请问这个参数是什么意思呢？

**Comment by leoleojie** [comment]
1、在计算损失时，CrossEntropyLoss的计算实际上等价于LogSoftmax及negative log-likelihood loss(NLLLoss)的组合，具体请参考[pytorch官方文档](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html?highlight=crossentropyloss#torch.nn.CrossEntropyLoss)
2、eval_step表示的是经过多少个epoch的训练之后在验证集上做一次验证，具体请参考伯乐文档[Traning Settings](https://recbole.io/docs/user_guide/config/training_settings.html?highlight=eval_step)

**Comment by youngzw** [comment]
好的，非常感谢
